ipns.go

```diff
@@ -25,6 +33,36 @@ func (s *Shell) Publish(node string, value string) error {
 	return nil
 }
 
+// PublishWithDetails is used for fine grained control over record publishing
+func (s *Shell) PublishWithDetails(contentHash, key string, lifetime, ttl time.Duration, resolve bool) (*PublishResponse, error) {
+	var pubResp PublishResponse
```

Comment:
I'd put it above/closer to `json.Unmarshal` line

Reply:
Done! :D